From 0cf0a88607fb7cfc7574d4d4b923fe36add82dd0 Mon Sep 17 00:00:00 2001
From: Michael Demeter <michael.demeter@intel.com>
Date: Fri, 11 Oct 2013 15:37:57 -0700
Subject: [PATCH 05/24] Smack enabled systems need /dev special devices
 correctly labeled

- Add AC_DEFINE for HAVE_SMACK to configure.ac
- Add Check for smack in Makefile.am to include smack default rules
- Add smack default rules to label /dev/xxx correctly for access

Change-Id: Iab07eb632b487b9ac4567cd08d0da6879709d44f
Signed-off-by: Michael Demeter <michael.demeter@intel.com>
---
 Makefile.am                       |  5 +++++
 configure.ac                      |  1 +
 rules/55-udev-smack-default.rules | 21 +++++++++++++++++++++
 3 files changed, 27 insertions(+)
 create mode 100644 rules/55-udev-smack-default.rules

diff --git a/Makefile.am b/Makefile.am
index 4028112..20bc6c9 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -3108,6 +3108,11 @@ dist_udevrules_DATA += \
 nodist_udevrules_DATA += \
 	rules/99-systemd.rules
 
+if HAVE_SMACK
+dist_udevrules_DATA += \
+	rules/55-udev-smack-default.rules
+endif
+
 dist_udevhwdb_DATA = \
 	hwdb/20-pci-vendor-model.hwdb \
 	hwdb/20-pci-classes.hwdb \
diff --git a/configure.ac b/configure.ac
index 18b7198..3c43134 100644
--- a/configure.ac
+++ b/configure.ac
@@ -633,6 +633,7 @@ AS_HELP_STRING([--with-smack-run-label=STRING],
 
 if test "x${have_smack}" = xyes ; then
         AC_DEFINE(HAVE_SMACK, 1, [Define if SMACK is available])
+        AM_CONDITIONAL([HAVE_SMACK], [true])
 fi
 
 # ------------------------------------------------------------------------------
diff --git a/rules/55-udev-smack-default.rules b/rules/55-udev-smack-default.rules
new file mode 100644
index 0000000..be8c4f8
--- /dev/null
+++ b/rules/55-udev-smack-default.rules
@@ -0,0 +1,21 @@
+# do not edit this file, it will be overwritten on update
+
+KERNEL=="null", SMACK="*"
+KERNEL=="zero", SMACK="*"
+KERNEL=="console", SMACK="*"
+KERNEL=="kmsg", SMACK="*"
+KERNEL=="video*", SMACK="*"
+KERNEL=="card*", SMACK="*"
+
+SUBSYSTEM=="graphics", GROUP="video", SMACK="*"
+SUBSYSTEM=="drm", GROUP="video", SMACK="*"
+SUBSYSTEM=="dvb", GROUP="video", SMACK="*"
+
+SUBSYSTEM=="tty", KERNEL=="ptmx", GROUP="tty", MODE="0666", SMACK="*"
+SUBSYSTEM=="tty", KERNEL=="tty", GROUP="tty", MODE="0666", SMACK="*"
+SUBSYSTEM=="tty", KERNEL=="tty[0-9]*", GROUP="tty", MODE="0620", SMACK="*"
+SUBSYSTEM=="vc", KERNEL=="vcs*|vcsa*", GROUP="tty", SMACK="*"
+KERNEL=="tty[A-Z]*[0-9]|pppox[0-9]*|ircomm[0-9]*|noz[0-9]*|rfcomm[0-9]*", GROUP="dialout", SMACK="*"
+
+SUBSYSTEM=="input", KERNEL=="mouse*|mice|event*", MODE="0640", SMACK="*"
+SUBSYSTEM=="input", KERNEL=="ts[0-9]*|uinput", MODE="0640", SMACK="*"
-- 
1.8.4.5

